src/settings/Api.ts

```diff
@@ -81,79 +94,72 @@ export class Api {
 		return axios.post(generateUrl('/apps/settings/') + '/settings/authorizedgroups/saveSettings', {
 			newGroups,
 			class: classname,
-		}).then((data) => data.data)
+		})
 	}
 
 	async createFolder(mountPoint: string): Promise<Folder> {
-		return $.post(this.getUrl('folders'), {
-			mountpoint: mountPoint
-		}, null, 'json').then((data: OCSResult<Folder>) => data.ocs.data)
+		return axios.post(this.getUrl('folders'), { mountpoint: mountPoint })
+			.then((response: AxiosOCSResult<Folder>) => response.data.ocs.data)
 	}
 
 	async deleteFolder(id: number): Promise<void> {
-		return $.ajax({
-			url: this.getUrl(`folders/${id}`),
-			type: 'DELETE'
-		})
+		return axios.delete(this.getUrl(`folders/${id}`))
 	}
 
 	async addGroup(folderId: number, group: string): Promise<void> {
-		return $.post(this.getUrl(`folders/${folderId}/groups`), {
-			group
+		return axios.post(this.getUrl(`folders/${folderId}/groups`), {
+			 group
 		})
 	}
 
 	async removeGroup(folderId: number, group: string): Promise<void> {
-		return $.ajax({
-			url: this.getUrl(`folders/${folderId}/groups/${group}`),
-			type: 'DELETE'
-		})
+		return axios.delete(this.getUrl(`folders/${folderId}/groups/${group}`))
 	}
 
 	async setPermissions(folderId: number, group: string, permissions: number): Promise<void> {
-		return $.post(this.getUrl(`folders/${folderId}/groups/${group}`), {
+		return axios.post(this.getUrl(`folders/${folderId}/groups/${group}`), {
 			permissions
 		})
 	}
 
 	async setManageACL(folderId: number, type: string, id: string, manageACL: boolean): Promise<void> {
-		return $.post(this.getUrl(`folders/${folderId}/manageACL`), {
+		return axios.post(this.getUrl(`folders/${folderId}/manageACL`), {
 			mappingType: type,
 			mappingId: id,
 			manageAcl: manageACL ? 1 : 0
 		})
 	}
 
 	async setQuota(folderId: number, quota: number): Promise<void> {
-		return $.post(this.getUrl(`folders/${folderId}/quota`), {
+		return axios.post(this.getUrl(`folders/${folderId}/quota`), {
 			quota
 		})
 	}
 
 	async setACL(folderId: number, acl: boolean): Promise<void> {
-		return $.post(this.getUrl(`folders/${folderId}/acl`), {
+		return axios.post(this.getUrl(`folders/${folderId}/acl`), {
 			acl: acl ? 1 : 0
 		})
 	}
 
 	async renameFolder(folderId: number, mountpoint: string): Promise<void> {
-		return $.post(this.getUrl(`folders/${folderId}/mountpoint`), {
+		return axios.post(this.getUrl(`folders/${folderId}/mountpoint`), {
 			mountpoint
 		})
 	}
 
 	async aclMappingSearch(folderId: number, search: string): Promise<{groups: ManageRuleProps[], users: ManageRuleProps[]}> {
-		return $.getJSON(this.getUrl(`folders/${folderId}/search?format=json&search=${search}`))
-			.then((data: OCSResult<{ groups: OCSGroup[]; users: OCSUser[]; }>) => {
+		return axios.get(this.getUrl(`folders/${folderId}/search`), { params: { search } })
```

Comment:
Why is this an async function but then no await is used but `.then()` chain?
(I know this is unrelated to your changes but no good code style)

Reply:
Right, it could be sync and still return the Promise as it already does.